### PR TITLE
MRG: updates to clean up docker builds and circle tests

### PR DESCRIPTION
**Pull request by satra**

### CODE REVIEW

**Comment by satra** [comment]
@yarikoptic, @jbpoline, @dnkennedy - this is ready for review and merge.

things changed:
- SimplePrep.sh now supports standalone check, install, and test
- Dockerfile template now uses this new script
- expected output regenerated with new dockerfile
- circle ci now builds the docker image to test.
- updated README
- older outputs moved to other_outputs

things todo once merged:
- build dockerhub dockerfile so the part of the readme that says you can use docker pull actually works
- right before release i will fix the versions of the simple_workflow repo downloaded in the script and rebuild dockerhub dockerfile

File: Dockerfile [hunk 1]
```diff
@@ -28,47 +23,45 @@ RUN apt-get install -y -q eatmydata
...
 # We might be just fine with the core here
 RUN bash -c "eatmydata apt-get install -y -q fsl-core fsl-first-data"
 
-RUN apt-get clean
+RUN apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /boot /media /mnt /srv
+
+ENV FSLMULTIFILEQUIT=TRUE
+ENV POSSUMDIR=/usr/share/fsl/5.0
+ENV LD_LIBRARY_PATH=/usr/lib/fsl/5.0
```
Comment (yarikoptic) [inline Dockerfile:37]: such hardcoding of paths wouldn't be robust for upgrades

File: Simple_Prep_docker [hunk 2]
```diff
@@ -66,11 +66,9 @@ sed-i "$dockerfile" -e '/DL_APT/d'
 tag_=simple_prep_${USER}_${DL_DIST}_${DL_DATE}
 tag=repronim:${tag_}
 echo "I: tag $tag"
-if docker images | grep -q repronim.*${tag_}; then
-    echo "I: tag $tag already exists -- skipping rebuilding"
-else
+if [ $(command -v docker) ]; then
     docker build -t $tag -f $dockerfile .
```
Comment (yarikoptic) [inline Simple_Prep_docker:70]: wouldn't it build and only then fail if tag already exists? (sorry didn't try, but I thought that my above check was to fail early for that reason)

File: Dockerfile (hunk 1 above)
Comment (satra) [reply Dockerfile:37]: for the moment i decided to resort to this, till we figure out a better thing that not only covers FSL from neurodebian, but also conda and pip, etc.,. i.e niceman. the only thing that can affect this right now, is if someone wanted to use FSL 4.

File: Simple_Prep_docker (hunk 2 above)
Comment (satra) [reply Simple_Prep_docker:70]: it should rebuild based on content of dockerfile and it's connected files and not tag. tag doesn't guarantee much if we have complicated dependencies.

**Review by yarikoptic — COMMENTED** [review]
so my main concern is hardcoding fsl paths which wouldn't allow to use it for next fsl release (whenever that one comes) even if it would be compatible

**Comment by satra** [comment]
@yarikoptic - ok i fixed it !

File: Simple_Prep.sh [hunk 3]
```diff
@@ -3,7 +3,13 @@
 set -e
 
 # Check if FSL is available
-[ ! $(command -v bet) ] && echo "No bet: Make sure an FSL version is installed and available." && exit 1
+if [ ! $(command -v bet) ]; then
```
Comment (yarikoptic) [inline Simple_Prep.sh]: cool.  didn't know about `command -v`... usually used `which` but that one is not fully portable iirc, so for those cases used 'hash' ;-)

**Pull request merged**